lib/Dialect/SV/Transforms/SVExtractTestCode.cpp

```diff
@@ -712,10 +708,21 @@ void SVExtractTestCodeImplPass::runOnOperation() {
 
       // Walk the module and add dead operations to `opsToErase`.
       op.walk([&](Operation *operation) {
-        if (&op != operation && !opsAlive.count(operation))
+        // Skip the module itself.
+        if (&op == operation)
+          return;
+
+        // Update `opsToErase`.
+        if (opsAlive.count(operation))
+          opsToErase.erase(operation);
```

Comment:
Note that opsToErase is `SmallPtrSetImpl` so it won't cause an issue like fixed by https://github.com/llvm/circt/pull/5353